utils.py

```diff
@@ -6,5 +15,109 @@ def filter_country_sort_year(df, country):
     :return: DataFrame
     """
     df = df.loc[df["country_name"] == country]
-    df = df.sort_values(["year"])
+
+    df = df[df.year >= START_YEAR]
+    df = df.sort_values(["year"], ascending=False)
     return df
+
+
+import math
+
+millnames = ["", " K", " M", " B", " T"]
+
+
+def millify(n):
+    n = float(n)
+    millidx = max(
+        0,
+        min(
+            len(millnames) - 1, int(math.floor(0 if n == 0 else math.log10(abs(n)) / 3))
+        ),
+    )
+
+    return "{:.2f}{}".format(n / 10 ** (3 * millidx), millnames[millidx])
+
+
+def handle_empty_plot(fig, text=None):
+    if not text:
+        text = "No Matching Data Found"
+    fig.update_layout(
+        annotations=[
+            dict(
+                text=text,
+                showarrow=False,
+                x=None,
+                y=None,
+            )
+        ],
+    )
+    return fig
+
+
+def get_percentage_change_text(percent):
+    if percent > 0:
+        return f"increased by {percent:.2f}%"
+    return f"decreased by {-1 * percent:.2f}%"
+
+
+def calculate_PCC(df, x_col, y_col):
+    """
+    Calculate the Pearson Correlation Coefficient between two columns
+    :param df: DataFrame
+    :param x_col: str
+    :param y_col: str
+    :return: float
+    """
+    df = df[[x_col, y_col]].dropna()
+    return pearsonr(df[x_col], df[y_col])[0]
+
+
+def get_correlation_text(df, x_col, y_col):
+    """
+    Get the correlation text based on the PCC value
+    :param df: DataFrame
+    :param x_col: str
+    :param y_col: str
+    :return: str
+    """
+    pcc = calculate_PCC(df, x_col, y_col)
+    if pcc > 0:
+        text = "positive and "
+    else:
+        text = "negative and "
+    abs_pcc = abs(pcc)
+
+    for threshold, pcc_text in CORRELATION_THRESHOLDS.items():
+        if abs_pcc < float(threshold):
+            text += pcc_text
+            break
+    return text + "(PCC={:.2f})".format(pcc)
```

Comment:
Is PCC a standard acronym? Otherwise we could say (r=) instead?

Reply:
hmm.. 
I think if we say PCC, we are also saying that the model is a simple linear regression. 
R^2 might be too general of a term (it does not imply the model form?)

Reply:
following the same pattern from 
https://github.com/weilu/rpf-country-dash/pull/4#discussion_r1742086914